src/main/java/com/google/api/gax/bundling/ThresholdBundler.java

```diff
@@ -52,26 +52,23 @@
  */
 public class ThresholdBundler<E> {
 
-  private ImmutableList<BundlingThreshold<E>> thresholds;
-  private ImmutableList<ExternalThreshold<E>> externalThresholds;
+  private ImmutableList<BundlingThreshold<E>> thresholdPrototypes;
+  private ImmutableList<ExternalThreshold<E>> externalThresholdPrototypes;
   private final Duration maxDelay;
 
   private final Lock lock = new ReentrantLock();
   private final Condition bundleCondition = lock.newCondition();
-  private boolean bundleReady = false;
-  private BundleHandle currentBundleHandle;
-
-  private Stopwatch bundleStopwatch;
-  private final List<E> data = new ArrayList<>();
+  private Bundle currentOpenBundle;
+  private List<Bundle> closedBundles = new ArrayList<>();
 
   private ThresholdBundler(ImmutableList<BundlingThreshold<E>> thresholds,
       ImmutableList<ExternalThreshold<E>> externalThresholds,
       Duration maxDelay) {
-    this.thresholds = copyResetThresholds(Preconditions.checkNotNull(thresholds));
-    this.externalThresholds = copyResetExternalThresholds(
+    this.thresholdPrototypes = copyResetThresholds(Preconditions.checkNotNull(thresholds));
+    this.externalThresholdPrototypes = copyResetExternalThresholds(
```

Comment:
They aren't really defaults, because they can't be overwritten with other values. They serve as the prototypes to copy from and create new thresholds when there is a new bundle.